packages-native/actions/src/camera/TakePictureAdvanced.ts

```diff
@@ -0,0 +1,213 @@
+// Other code you write will be lost the next time you deploy the project.
+
+import { Alert, Linking } from "react-native";
+import ImagePicker from "react-native-image-picker";
+
+type PictureSource = "camera" | "imageLibrary" | "either";
+
+type PictureQuality = "original" | "low" | "medium" | "high" | "custom";
+
+/**
+ * @param {MxObject} picture - This field is required.
+ * @param {"NativeMobileResources.PictureSource.camera"|"NativeMobileResources.PictureSource.imageLibrary"|"NativeMobileResources.PictureSource.either"} pictureSource - Select a picture from the library or the camera. The default is to let the user decide.
+ * @param {"NativeMobileResources.PictureQuality.original"|"NativeMobileResources.PictureQuality.low"|"NativeMobileResources.PictureQuality.medium"|"NativeMobileResources.PictureQuality.high"|"NativeMobileResources.PictureQuality.custom"} pictureQuality - Set to empty to use default value 'medium'.
+ * @param {Big} maximumWidth - The picture will be scaled to this maximum pixel width, while maintaing the aspect ratio.
+ * @param {Big} maximumHeight - The picture will be scaled to this maximum pixel height, while maintaing the aspect ratio.
+ * @param {MxObject} pictureData - Additional info about the picture will be stored in this object. Create it before calling this action.
+ * @returns {Promise.<boolean>}
+ */
+export async function TakePicture(
+    picture?: mendix.lib.MxObject,
+    pictureSource?: PictureSource,
+    pictureQuality?: PictureQuality,
+    maximumWidth?: BigJs.Big,
+    maximumHeight?: BigJs.Big,
+    pictureData?: mendix.lib.MxObject
+): Promise<boolean> {
+    // BEGIN USER CODE
+    // Documentation https://github.com/react-native-community/react-native-image-picker/blob/master/docs/Reference.md
+
+    if (!picture) {
+        return Promise.reject(new Error("Input parameter 'Picture' is required"));
+    }
+
+    if (!picture.inheritsFrom("System.FileDocument")) {
+        const entity = picture.getEntity();
+        return Promise.reject(new Error(`Entity ${entity} does not inherit from 'System.FileDocument'`));
+    }
+
+    if (pictureQuality === "custom" && !maximumHeight && !maximumWidth) {
+        return Promise.reject(
+            new Error("Picture quality is set to 'Custom', but no maximum width or height was provided")
+        );
+    }
+
+    return takePicture()
+        .then((response: any) => {
+            if (!response || !response.uri) {
+                return false;
+            }
+            if (pictureData) {
+                pictureData.set("URI", response.uri);
+                pictureData.set("IsVertical", response.isVertical);
+                pictureData.set("Width", response.width);
+                pictureData.set("Height", response.height);
+                pictureData.set("FileSize", response.fileSize);
+                pictureData.set("FileType", response.type);
+            }
+            return storeFile(picture, response.uri);
```

Comment:
I think is also valid to attache the filename inside the MXObject

Reply:
Filename was empty when I tested.

Reply:
You are using filename in the line 101, so just reuse it

Reply:
You're right, that bit I did not touch, but if it works there it should work anywhere in the code. I will first test in my test project and then change the code here.

Reply:
That filename is extracted from the uri.

Reply:
I added it. Only filled on Android in the response object. I mentioned that in the documentation of the attribute in the entity.

Reply:
Is there any particular case that only android can handle it?

Reply:
It seems only on Android fileName has a value. On iOS it is empty.